rest/api_collections_test.go

```diff
@@ -104,6 +105,65 @@ func TestCollectionsPutDocInKeyspace(t *testing.T) {
 	}
 }
 
+// TestCollectionsPublicChannel ensures that a doc routed to the public channel is accessible by a user with no other access.
+func TestCollectionsPublicChannel(t *testing.T) {
+	tb := base.GetTestBucket(t)
+	defer tb.Close()
+
+	ds := tb.GetSingleDataStore()
+	dataStoreName, ok := base.AsDataStoreName(ds)
+	require.True(t, ok)
+	keyspaceName := "db" + "." + dataStoreName.ScopeName() + "." + dataStoreName.CollectionName()
+
+	const (
+		username = "alice"
+		password = "pass"
+	)
+
+	rt := NewRestTester(t, &RestTesterConfig{
+		CustomTestBucket: tb.NoCloseClone(),
+		DatabaseConfig: &DatabaseConfig{
+			DbConfig: DbConfig{
+				Users: map[string]*auth.PrincipalConfig{
+					username: {Password: base.StringPtr(password)},
+				},
+				Scopes: ScopesConfig{
+					dataStoreName.ScopeName(): ScopeConfig{
+						Collections: map[string]CollectionConfig{
+							dataStoreName.CollectionName(): {},
+						},
+					},
+				},
```

Comment:
I don't think you need this code or a custom bucket at all here.

Reply:
Copy pasted from the test above - but yeah the new helpers may make this a lot smaller. I'll take a look at using them